micropy/project/modules/packages.py

```diff
@@ -138,28 +118,29 @@ def add_package(self, package, dev=False, **kwargs):
             dict: Dictionary of packages
 
         """
-        pkg = package
-        if isinstance(package, str):
-            pkg = next(requirements.parse(package))
+
+        source = create_dependency_source(package)
+        pkg = source.package
         self.log.info(f"Adding $[{pkg.name}] to requirements...")
-        specs = "".join(next(iter(pkg.specs))) if pkg.specs else "*"
         if self.packages.get(pkg.name, None):
-            self.log.error(f"$[{pkg.name}] is already installed!")
+            self.log.error(f"$[{pkg}] is already installed!")
             self.update()
             return None
-        self.packages[pkg.name] = specs
-        self.parent.config.set(f'{self.name}.{pkg.name}', specs)
+        self.packages[pkg.name] = pkg.pretty_specs
         try:
             self.load()
-        except JSONDecodeError:
+        except ValueError:
             self.log.error(f"Failed to find package $[{pkg.name}]!")
             self.log.error("Is it available on PyPi?")
             self.packages.pop(pkg.name)
+            self.parent.config.pop(f"{self.name}.{pkg}")
         except Exception as e:
-            self.log.error(f"An unknown error occured during the installation of $[{pkg.name}]!",
-                           exec=e)
+            self.log.error(f"An error occured during the installation of $[{pkg.name}]!")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Instance of 'PackagesModule' has no 'log' member](https://app.codacy.com/app/BradenM/micropy-cli/pullRequest?prid=4593453)